### DISCUSSION TITLE
Topic on webservice

### DISCUSSION THREAD

**Discussion by gsportelli73** [YetiForceCompany/YetiForceCRM#14886] [How to]
Problem webservice 401 on login in CRM.
Description:
My crm doesn't accept login via webservice. 
I use webservice to integrate other software in CRM. 
After a lot of day i understand that problem is apache configuration. 
Apache uses some method to implement php:
1- Php wiht lspi module
2- php via mod_cgi
3- php via mod_fgid
4- php via dso module
5- php via suphp e suexec.
When you set some method number 2,4 htaccess is ignored from Apache in the part of rewrite httpd header. 
I discover suphp that resolve this problem.
I write this to storage a small note !

**Comment by mariuszkrzaczkowski** [comment]
use no rewrite mode
```
rewrite
CRM_URL/webservice/RestApi/Users/Login
CRM_URL/webservice/RestApi/Accounts/RecordRelatedList/117/Contacts
without rewrite
CRM_URL/webservice.php?_container=RestApi&module=Users&action=Login
CRM_URL/webservice.php?_container=RestApi&module=Accounts&action=RecordRelatedList&record=117&param=Contacts
```

**Reply by mariuszkrzaczkowski** [reply]
https://doc.yetiforce.com/api/?urls.primaryName=RestApi

**Comment by gsportelli73** [comment]
Thank you
I now updated on latest yetiforce, i have tuned apache e php for it.
Untile last update 6.1.2 ?
My websrice it was ok now no.
But must i pay service ?
Infact now the error it is 404.
Thank for attention

Il giorno sab 3 lug 2021 alle ore 20:28 Mariusz Krzaczkowski <
***@***.***> ha scritto:

> https://doc.yetiforce.com/api/?urls.primaryName=RestApi
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/YetiForceCompany/YetiForceCRM/discussions/14886#discussioncomment-961203>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AGTIFAQ7J4CQFUKJBVLUN53TV5JE3ANCNFSM47X4FIXA>
> .
>


-- 
Giuseppe Sportelli
Consulenze Informatiche
Formazione Informatica

**Reply by mariuszkrzaczkowski** [reply]
RestAPI - free
Portal - paid

The Portal API from 6.1 is RestAPI in 6.2

If you do not know something, ask, moods and resentments will not help anyone
